internal/controllers/topology/cluster/patches/variables/variables.go

```diff
@@ -182,6 +182,16 @@ func MachineDeployment(mdTopology *clusterv1.MachineDeploymentTopology, md *clus
 	if md.Spec.Replicas != nil {
 		builtin.MachineDeployment.Replicas = ptr.To[int64](int64(*md.Spec.Replicas))
 	}
+	if mdTopology.Metadata.Labels != nil || mdTopology.Metadata.Annotations != nil {
```

Comment:
Let's please take the labels & annotations from md not from mdTopology.

The ones from md contain *all* labels & annotations that we eventually set on the MachineDeployment